### DISCUSSION TITLE
using actionMeta in a send action

### DISCUSSION THREAD

**Discussion by Manborough** [statelyai/xstate#3000] [Q&A]
Is there any way to access the actionMeta in a send action? I can't seem to get it to work.

```js
{states, context, ect},
{
  actions: {
    example_action: (context, event, actionMeta) => {
	  	  // How to acommplish this?
		  send({type: 'test', foo: actionMeta.action.bar })
    }
  }
}
```

**Comment by davidkpiano** [comment]
`send` is not an imperative function; it returns an action object.

What are you trying to accomplish? There's probably a straightforward way.

**Reply by Manborough** [reply]
I've got an action to send an update to an actor which is written inline
```js
entry: send({ type: 'UPDATE_PARENT_STATE', parent_state: "fooBar" }, { to: (context) => context.consultant }),

entry: send({ type: 'UPDATE_PARENT_STATE', parent_state: "HelloWorld" }, { to: (context) => context.consultant }),
```
There's a few places this is used but with the parent state changing between them
I thought I could try and refactor these actions into the actions object and try to deduplicate it.

```js
example_state: {
    entry: { type: 'update_parent_state', parent_state: "fooBar" },
}

example_state: {
    entry: { type: 'update_parent_state', parent_state: "HelloWorld" },
}
```
So in the actions object I tried this but quickly realised that actionMeta isn't passed to send.
```js
actions: {
    example_action: send((context, event, actionMeta) => {
        {type: 'test', parent_state: actionMeta.action.parent_state }
   } )
}
```

**Reply by Manborough** [reply]
This might be a more straightforward example.

If I had a service which wanted to report back errors to the parent how could I put that into an action that isn't inline.

```js
state1: {
  on: { FOO: "state2" actions: { type: "send_parent_error_message", error_message: "This action failed!" }} }
state2: {
  on: { FOO: "state3" actions: { type: "send_parent_error_message", error_message: "This action failed too!" } 
},
state3: {}
```

I'd like to send that error message through to the parent using a single action.

```js
actions: {
    send_parent_error_message: sendParent({type: "SET_ERROR_MESSAGE", error_message: "", })
}
```
